From 1556346238ebc87fef4f98b132171dfd4461571a Mon Sep 17 00:00:00 2001
From: SGCMarkus <markusornik@gmail.com>
Date: Sun, 9 Oct 2022 16:12:50 +0200
Subject: [PATCH 2/2] Camera: Avoid roundBufferDimensionsNearest also for
 HAL_PIXEL_FORMAT_BLOB

This fixes the Ultra-Res feature with Motorolas MotCamera3
It uses multiple camera streams, with only the blob format using the
full sensor size (e.g. on berlin: 12000 x 9000 for 108 MP)

Change-Id: I025e068b2d3ff1123599ae5655655876ee28f39b
---
 .../libcameraservice/utils/SessionConfigurationUtils.cpp   | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp b/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp
index b49f86c0e7..eb45fbec26 100644
--- a/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp
+++ b/services/camera/libcameraservice/utils/SessionConfigurationUtils.cpp
@@ -213,10 +213,11 @@ bool roundBufferDimensionNearest(int32_t width, int32_t height,
     // Avoid roundBufferDimensionsNearest for privileged client YUV streams to meet the AIDE2
     // requirement. AIDE2 is vendor enhanced feature which requires special resolutions and
     // those are not populated in static capabilities.
-    if (isPriviledgedClient == true && format == HAL_PIXEL_FORMAT_YCbCr_420_888) {
+    if (isPriviledgedClient == true &&
+        (format == HAL_PIXEL_FORMAT_YCbCr_420_888 || format == HAL_PIXEL_FORMAT_BLOB)) {
         ALOGI("Bypass roundBufferDimensionNearest for privilegedClient YUV streams "
-              "width %d height %d",
-              width, height);
+              "width %d height %d for format %d",
+              width, height, format);
 
         bestWidth  = width;
         bestHeight = height;
-- 
2.34.1

